app/models/study.rb

```diff
@@ -553,6 +556,19 @@ def rebroadcast
     broadcast
   end
 
+  # Returns the PolyMetadatum object associated with the given key.
+  #
+  # @param key [String] The key of the PolyMetadatum to find.
+  #
+  # @return [PolyMetadatum, nil] The PolyMetadatum object with the given key,
+  #   or nil if no such PolyMetadatum exists.
+  #
+  # @example
+  #   study.poly_metadatum_by_key("sample_key")
+  def poly_metadatum_by_key(key)
+    poly_metadata.find { |pm| pm.key == key.to_s }
```

Comment:
just checking, this works without having the metadatable_id (the study_id here) in the find parameters? its finding it through the has_many relationship at the top of the class?

Reply:
Yes, it is because of the has_many relationship defined as the following. I have copied the same setup with Request metadata.
```
# Can have many key value pairs of metadata
has_many :poly_metadata, as: :metadatable, dependent: :destroy
```

The poly_metadata collection provides access to the PolyMetadatums that belong to the Study only. The poly_metadatum_by_key(key) method finds the one with the specified key. PolyMetadatum has already metadatable_id as the Study's id and metadatable_type as the Study's type from creation.